src/components/widgets/GoogleAuth.vue

```diff
@@ -1,26 +1,82 @@
 <template>
-  <button @click="handleUserGoogleAuthorization">Login Using Google</button>
+  <div class="google-auth">
+    <div class="google-auth__body">
+      <el-button class="background w-100" type="primary" @click="handleUserGoogleAuthorize">
+        Войти с помощью Google
+        <icon-template class="icon-20 ml-10" name="google" />
+      </el-button>
+    </div>
+
+    <base-dialog v-model="isRegistrationDialogVisible">
+      <h2 class="ta-c">Введите свои данные</h2>
+      <google-registration-form class="mt-30" />
+    </base-dialog>
+  </div>
 </template>
 
 <script lang="ts" setup>
 import { googleAuthCodeLogin } from 'vue3-google-login';
-import { authApi } from '@/api/auth/auth.api';
-import { useAuthStore } from '@/stores/authStore';
+import { GoogleRegistrationFormType } from '@/types/authFormTypes';
+import { provide, ref } from 'vue';
+import GoogleRegistrationForm from '@/components/Forms/GoogleRegistrationForm.vue';
+import { authApi } from '@/api/KY/AuthService/auth.api';
+import { useTokens } from '@/composables/useTokens';
 
-const authStore = useAuthStore();
+const isRegistrationDialogVisible = ref(false);
 
-const handleUserGoogleAuthorization = async () => {
+const handleUserGoogleAuthorize = async () => {
   const googleData = await googleAuthCodeLogin();
-  const [error, data] = await authApi.authGoogleUser(googleData.code);
+  const [error, data] = await authApi.loginGoogleUser(googleData.code);
 
   if (!error && data) {
-    const { access, refresh } = data;
+    const { is_register, access, refresh } = data;
+
+    const { setAccess, setRefresh } = useTokens(localStorage);
+
+    setRefresh(refresh);
+
+    setAccess(access);
+
+    if (!is_register) {
+      isRegistrationDialogVisible.value = true;
+    }
+  }
+};
+
+const handleUserGoogleRegister = async (form: GoogleRegistrationFormType) => {
```

Comment:
```suggestion
const handleUserGoogleRegister = async (form: GoogleRegistrationFormType): void => {
```

И просмотри и лучше во всех местах добавить тип возвращаемого значения